src/processes.c

```diff
@@ -1574,42 +1572,58 @@ static char *ps_get_cmdline(long pid, char *name, char *buf, size_t buf_len) {
   return buf;
 } /* char *ps_get_cmdline (...) */
 
-static int read_fork_rate(void) {
-  FILE *proc_stat;
-  char buffer[1024];
+static int read_fork_rate(const char *buffer) {
   value_t value;
-  bool value_valid = 0;
+  char id[] = "processes ";
+  char *processes;
 
-  proc_stat = fopen("/proc/stat", "r");
-  if (proc_stat == NULL) {
-    ERROR("processes plugin: fopen (/proc/stat) failed: %s", STRERRNO);
+  int status;
+  char *fields[2];
+  int fields_num;
+
+  processes = strstr(buffer, id);
+  if (!processes) {
+    WARNING("processes not found");
```

Comment:
```suggestion
    WARNING("'processes ' not found in /proc/stat");
```
Maybe a bit uglier, but conveys more info.

If you think it is better like that, there are other occurrences of the same kind of warning.